ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -7983,7 +7984,17 @@ protected boolean enableColumnStatsCollecting() {
     return true;
   }
 
-  private Path getCtasLocation(CreateTableDesc tblDesc) throws SemanticException {
+  private Path getCtasLocation(CreateTableDesc tblDesc, boolean createTableWithSuffix) throws SemanticException {
+    Path destinationPath = getCtasLocationWithoutSuffix(tblDesc);
+    if (createTableWithSuffix) {
+      long txnId = ctx.getHiveTxnManager().getCurrentTxnId();
+      String suffix = AcidUtils.getPathSuffix(txnId);
+      destinationPath = new Path(destinationPath.toString() + suffix);
+    }
+    return destinationPath;
+  }
+
+  private Path getCtasLocationWithoutSuffix(CreateTableDesc tblDesc) throws SemanticException {
```

Comment:
please keep it as `getCtasLocation` and call overloaded getCtasLocation(tblDesc, false)

Reply:
Done